### ISSUE TITLE
Index of conditional query hits returns a Tire index opposed to an enumerator

### ISSUE THREAD

**Issue by krigar** [karmi/retire#574]
This happens when upgrading from Rails 3.2.5 to 3.2.6

```
users = User.where('sign_in_count >= 0')
users.index
=> #<Tire::Index:0x41a0ae0 @name="users">
```

If you remove the condition the expected enumerator is returned:

```
users = User.all
users.index
=> #<Enumerator: [#<User id: 1 ...
```

I don't know if the bug lies within Rails or Tire, but I hope this is not intended behavior. Such overloading can prove dangerous and extremely hard to debug.

**Comment by karmi** [comment]
@krigar, first thing, _Tire_ actively encourages you to use the `tire` proxy if you're concerned about method clashes, see Readme:

> So, instead of writing Article.search, you could write Article.tire.search, and instead of @article.update_index you could write @article.tire.update_index, to be on the safe side.

_Tire_ brings its methods into the model only when they aren't [already defined](https://github.com/karmi/tire/blob/master/lib/tire/model/search.rb#L296-L312). 

I can't see any evidence of the `index` method for ActiveRecord::Relation?

``` ruby
$ rails console
> pry
> Rails.version
=> "3.2.11"
> Article.where(title: 'Title 1').class
=> ActiveRecord::Relation
> find-method 'index'
No Methods Matched
```

**Comment by krigar** [comment]
Yeah, I eventually found the code where you assign the class methods and saw that you left the existing ones as they were, so then it had to be rails that was acting differently.

Turns out that in versions previous to 3.2.6 one could call index directly on ActiveRecord::Relations, even though it wasn't mentioned in the API. I've tried googling for it and checked various mail lists, but I haven't found anyone mention that it doesn't work any more post 3.2.6, so I suspect I was the only one on the planet that was using the index method directly on Relation without converting to an array first.
